packages/arb-token-bridge-ui/src/components/TransferPanel/Routes/ArbitrumRoute.tsx

```diff
@@ -0,0 +1,280 @@
+import { useNetworks } from '../../../hooks/useNetworks'
+import { useNetworksRelationship } from '../../../hooks/useNetworksRelationship'
+import { constants, utils } from 'ethers'
+import { Route, Token } from './Route'
+import { useAmountBigNumber } from '../hooks/useAmountBigNumber'
+import {
+  UseGasSummaryResult,
+  useGasSummary
+} from '../../../hooks/TransferPanel/useGasSummary'
+import {
+  NativeCurrency,
+  useNativeCurrency
+} from '../../../hooks/useNativeCurrency'
+import { CommonAddress } from '../../../util/CommonAddressUtils'
+import {
+  getOrbitDepositDuration,
+  getStandardDepositDuration,
+  getWithdrawalDuration
+} from '../../../hooks/useTransferDuration'
+import { isNetwork } from '../../../util/networks'
+import dayjs from 'dayjs'
+import { useSelectedToken } from '../../../hooks/useSelectedToken'
+import { isTokenNativeUSDC } from '../../../util/TokenUtils'
+import { useRouteStore } from '../hooks/useRouteStore'
+import { useMemo } from 'react'
+import { ERC20BridgeToken } from '../../../hooks/arbTokenBridge.types'
+
+const commonUsdcToken = {
+  decimals: 6,
+  address: CommonAddress.Ethereum.USDC
+}
+
+const bridgedUsdcToken = {
+  ...commonUsdcToken,
+  name: 'Bridged USDC',
+  symbol: 'USDC.e',
+  l2Address: CommonAddress.ArbitrumOne['USDC.e']
+}
+
+const nativeUsdcToken = {
+  ...commonUsdcToken,
+  name: 'USDC',
+  symbol: 'USDC',
+  l2Address: CommonAddress.ArbitrumOne.USDC
+}
+
+function getDuration({
+  isTestnet,
+  sourceChainId,
+  isTeleportMode,
+  isWithdrawal,
+  isOrbitChain
+}: {
+  isTestnet: boolean
+  sourceChainId: number
+  isTeleportMode: boolean
+  isWithdrawal: boolean
+  isOrbitChain: boolean
+}) {
+  if (isTeleportMode) {
+    return (
+      getStandardDepositDuration(isTestnet) + getOrbitDepositDuration(isTestnet)
+    )
+  }
+
+  if (isWithdrawal) {
+    return getWithdrawalDuration({
+      createdAt: dayjs().valueOf(),
+      sourceChainId: sourceChainId
+    })
+  }
+
+  if (isOrbitChain) {
+    return getOrbitDepositDuration(isTestnet)
+  }
+
+  return getStandardDepositDuration(isTestnet)
+}
+
+function getGasCostAndToken({
+  childChainNativeCurrency,
+  parentChainNativeCurrency,
+  gasSummaryStatus,
+  estimatedChildChainGasFees,
+  estimatedParentChainGasFees,
+  isDepositMode,
+  selectedToken
+}: {
+  childChainNativeCurrency: NativeCurrency
+  parentChainNativeCurrency: NativeCurrency
+  gasSummaryStatus: UseGasSummaryResult['status']
+  estimatedChildChainGasFees: UseGasSummaryResult['estimatedChildChainGasFees']
+  estimatedParentChainGasFees: UseGasSummaryResult['estimatedParentChainGasFees']
+  isDepositMode: boolean
+  selectedToken: ERC20BridgeToken | null
+}): {
```

Comment:
This function is accepting lots of parameters rather than being a hook with lots of internal calls to hook to make it easier to test